model/jpa/src/main/java/org/keycloak/connections/jpa/util/JpaUtils.java

```diff
@@ -56,11 +61,22 @@ public static String getTableNameForNativeQuery(String tableName, EntityManager
         return (schema==null) ? tableName : dialect.openQuote() + schema + dialect.closeQuote() + "." + tableName;
     }
 
+    private static List<ParsedPersistenceXmlDescriptor> transformPersistenceUnits(Collection<PersistenceUnitDescriptor> descriptors) {
+        return descriptors.stream().map(descriptor -> (ParsedPersistenceXmlDescriptor) descriptor).collect(Collectors.toList());
+    }
+
     public static EntityManagerFactory createEntityManagerFactory(KeycloakSession session, String unitName, Map<String, Object> properties, boolean jta) {
         PersistenceUnitTransactionType txType = jta ? PersistenceUnitTransactionType.JTA : PersistenceUnitTransactionType.RESOURCE_LOCAL;
-        List<ParsedPersistenceXmlDescriptor> persistenceUnits = new ArrayList<>(PersistenceXmlParser.locatePersistenceUnits(properties));
+        PersistenceXmlParser parser = PersistenceXmlParser.create(properties);
+        List<URL> urls = parser.getClassLoaderService().locateResources("META-INF/persistence.xml");
 
-        persistenceUnits.add(PersistenceXmlParser.locateIndividualPersistenceUnit(JpaUtils.class.getClassLoader().getResource("default-persistence.xml")));
+        List<ParsedPersistenceXmlDescriptor> persistenceUnits = urls.isEmpty() ? new ArrayList<>() : transformPersistenceUnits(parser.parse(urls).values());
```

Comment:
Mimic the behavior of the [PersistenceXmlParser#locatePersistenceUnits](https://github.com/hibernate/hibernate-orm/blob/6.6.19/hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/PersistenceXmlParser.java#L58) with the new approach for locating all these `persistence.xml` resources - inspired by the [HibernateOrmProcessor#parsePersistenceXmlDescriptors](https://github.com/quarkusio/quarkus/blob/main/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java#L269) in Quarkus.